File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        arr = [0] * 26
        beg, maxLen, largestCnt = 0,0,0
        for end in range(len(s)):
            arr[ord(s[end]) - ord('A')] += 1
            largestCnt = max(largestCnt, arr[ord(s[end]) - ord('A')])
            
            if (end - beg + 1 - largestCnt) > k:
                arr[ord(s[beg]) - ord('A')] -= 1
                beg += 1
            
            maxLen = max(maxLen, (end-beg+1))
            
        return maxLen
        

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] arr = new int[26];
        int beg = 0, maxLen = 0, largestCnt = 0;
        
        for(int end = 0; end < s.length(); end++) {
            arr[s.charAt(end) - 'A']++;
            largestCnt = Math.max(largestCnt, arr[s.charAt(end) - 'A']);
            
            if(end - beg + 1 - largestCnt > k) {
                arr[s.charAt(beg++) - 'A'] --; 
                System.out.println(Arrays.toString(arr));
            }
            
            maxLen = Math.max(maxLen, end-beg+1);
        }
        return maxLen;
        
    }
}